proxy/src/console/provider/neon.rs

```diff
@@ -300,23 +312,61 @@ impl super::Api for Api {
             .wake_compute_endpoint_rate_limiter
             .check(user_info.endpoint.normalize_intern(), 1)
         {
-            info!(key = &*key, "found cached compute node info");
             return Err(WakeComputeError::TooManyConnections);
         }
 
-        let mut node = permit.release_result(self.do_wake_compute(ctx, user_info).await)?;
-        ctx.set_project(node.aux.clone());
-        let cold_start_info = node.aux.cold_start_info;
-        info!("woken up a compute node");
+        let node = permit.release_result(self.do_wake_compute(ctx, user_info).await);
+        match node {
+            Ok(mut node) => {
+                ctx.set_project(node.aux.clone());
+                let cold_start_info = node.aux.cold_start_info;
+                debug!("woken up a compute node");
+
+                // store the cached node as 'warm'
+                node.aux.cold_start_info = ColdStartInfo::WarmCached;
+                let (_, cached) = self
+                    .caches
+                    .node_info
+                    .insert_unit(key.clone(), Ok(node.clone()));
+                node.aux.cold_start_info = cold_start_info;
```

Comment:
Why is the info in the cache different that the returned value? Maybe add a comment? (Yes, the code was there before, but does not make it clear what is happening here.)

Reply:
There is a comment! We want to store the cold_start_info as `warm_cached` so that when we retrieve the node from the cache, it shows up as cached in our metrics. Now I look at this code a bit more I want to refactor it though

Reply:
The comment states that we are doing it. That is actually visible in the code :). It is unclear why we are doing it, though.